src/overlays/actors/ovl_En_Ex_Item/z_en_ex_item.c

```diff
@@ -21,39 +44,477 @@ const ActorInit En_Ex_Item_InitVars = {
+                    this->actor.shape.unk_08 = 10.0f;
+                }
+                break;
+            case EXITEM_PURPLE_RUPEE_BOWLING:
+            case EXITEM_PURPLE_RUPEE_COUNTER:
+                this->unk_17C = func_8002EBCC;
+                this->unk_180 = func_8002ED80;
+                this->drawItemId = 0x70;
+                this->timer = 65;
+                this->prizeRotateTimer = 35;
+                this->scale = 0.5f;
+                this->unkFloat = 0.5f;
+                if (this->type == EXITEM_PURPLE_RUPEE_BOWLING) {
+                    this->actionFunc = EnExItem_BowlPrize;
+                } else {
+                    this->actionFunc = EnExItem_SetupBowlCounter;
+                    this->actor.shape.unk_08 = 40.0f;
+                }
+                break;
+            case EXITEM_GREEN_RUPEE_CHEST:
+            case EXITEM_BLUE_RUPEE_CHEST:
+            case EXITEM_RED_RUPEE_CHEST:
+            case EXITEM_13:
+            case EXITEM_14:
+                this->unk_17C = func_8002EBCC;
+                this->unk_180 = func_8002ED80;
+                this->timer = 7;
+                this->scale = 0.5f;
+                this->unkFloat = 0.5f;
+                this->actor.velocity.y = 10.0f;
+                switch (this->type) {
+                    case EXITEM_GREEN_RUPEE_CHEST:
+                        this->drawItemId = 0x6C;
+                        break;
+                    case EXITEM_BLUE_RUPEE_CHEST:
+                        this->drawItemId = 0x6D;
+                        break;
+                    case EXITEM_RED_RUPEE_CHEST:
+                        this->drawItemId = 0x6E;
+                        break;
+                    case EXITEM_14:
+                        this->drawItemId = 0x70;
+                        break;
+                }
+                this->actionFunc = EnExItem_ExitChest;
+                break;
+            case EXITEM_MAGIC_FIRE:
+            case EXITEM_MAGIC_WIND:
+            case EXITEM_MAGIC_DARK:
+                this->scale = 0.35f;
+                this->actionFunc = EnExItem_FairyMagic;
+                break;
+            case EXITEM_BULLET_BAG:
+                this->unk_17C = func_8002EBCC;
+                this->drawItemId = 0x6B;
+                this->scale = 0.1f;
+                this->timer = 80;
+                this->prizeRotateTimer = 35;
+                this->actionFunc = EnExItem_TargetPrizeApproach;
+                break;
+            case EXITEM_SMALL_KEY_CHEST:
+                break;
+        }
+    }
+}
+
+void EnExItem_BowlPrize(EnExItem* this, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 tmpf1;
+    f32 tmpf2;
+    f32 tmpf3;
+    f32 tmpf4;
+    f32 tmpf5;
+    f32 tmpf6;
+    f32 tmpf7;
+    f32 sp3C;
+
+    if (!this->stopRotate) {
+        this->actor.shape.rot.y += 0x1000;
+        if ((this->prizeRotateTimer == 0) && ((this->actor.shape.rot.y & 0xFFFF) == 0x9000)) {
```

Comment:
Is the mask really necessary here? (I'd imagine it comes from the += right before)
```suggestion
        if ((this->prizeRotateTimer == 0) && (this->actor.shape.rot.y == 0x9000)) {
```

Reply:
Can't get it to work without that. Note that shape.rot.y is a `s16`, so it wouldn't cast with `&0xFFFF`. That said, even using a `(u16)` cast doesn't work. It seems to have to be the `&0xFFFF`.

Reply:
Right, I forgot these were `s16`s, nvm